src/seedu/addressbook/data/person/DateOfBirth.java

```diff
@@ -0,0 +1,82 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+
+public class DateOfBirth {
+
+    public static final String EXAMPLE = "1.2.1993";
+    public static final String MESSAGE_DOB_CONSTRAINTS = "Numbers in date of birth should be in between periods and should be valid day, month and year.";
+    public static final String DOB_VALIDATION_REGEX = "\\S+\\.\\S+\\.\\S+";
+
+    public final String value;
+    private boolean isPrivate;
+
+    /**
+     * Validates given name.
+     *
+     * @throws IllegalValueException if given name string is invalid.
+     */
+    public DateOfBirth(String dob, boolean isPrivate) throws IllegalValueException {
+        this.isPrivate = isPrivate;
+        String trimmedName = dob.trim();
+        if (!isValidDOB(trimmedName)) {
+            throw new IllegalValueException(MESSAGE_DOB_CONSTRAINTS);
+        }
+        this.value = trimmedName;
+    }
+
+    /**
+     * Returns true if the given string is a valid person email.
+     */
+    public static boolean isValidDOB(String test) {
+
+        return test.matches(DOB_VALIDATION_REGEX) && isDateOfValidDayandMonth(test);
+    }
+
+    /**
+     * Check if the date of birth is valid.
```

Comment:
Should be `Checks`. Minor thing, but try to stick to coding standards as much as possible.